From 1382f3bfd47a1e4466b7fd54b57b113f5f58b823 Mon Sep 17 00:00:00 2001
From: Adrien Plazas <kekun.plazas@laposte.net>
Date: Tue, 13 Aug 2019 18:00:26 +0200
Subject: [PATCH 18/23] gtkfilechooserdialog: Drop the custom save entry

This avoids overloading the headebar, and lets the save entry be
populated just below it, where there is room.

Fixes https://source.puri.sm/Librem5/Apps_Issues/issues/99.
---
 gtk/gtkfilechooserdialog.c | 55 --------------------------------------
 1 file changed, 55 deletions(-)

--- a/gtk/gtkfilechooserdialog.c
+++ b/gtk/gtkfilechooserdialog.c
@@ -244,8 +244,6 @@
 static void response_cb (GtkDialog *dialog,
                          gint       response_id);
 
-static void setup_save_entry (GtkFileChooserDialog *dialog);
-
 G_DEFINE_TYPE_WITH_CODE (GtkFileChooserDialog, gtk_file_chooser_dialog, GTK_TYPE_DIALOG,
                          G_ADD_PRIVATE (GtkFileChooserDialog)
                          G_IMPLEMENT_INTERFACE (GTK_TYPE_FILE_CHOOSER,
@@ -454,9 +452,6 @@
 gtk_file_chooser_dialog_notify (GObject    *object,
                                 GParamSpec *pspec)
 {
-  if (strcmp (pspec->name, "action") == 0)
-    setup_save_entry (GTK_FILE_CHOOSER_DIALOG (object));
-
   if (G_OBJECT_CLASS (gtk_file_chooser_dialog_parent_class)->notify)
     G_OBJECT_CLASS (gtk_file_chooser_dialog_parent_class)->notify (object, pspec);
 }
@@ -512,55 +507,6 @@
 }
 
 static void
-setup_save_entry (GtkFileChooserDialog *dialog)
-{
-  gboolean use_header;
-  GtkFileChooserAction action;
-  gboolean need_entry;
-  GtkWidget *header;
-
-  g_object_get (dialog,
-                "use-header-bar", &use_header,
-                "action", &action,
-                NULL);
-
-  if (!use_header)
-    return;
-
-  header = gtk_dialog_get_header_bar (GTK_DIALOG (dialog));
-
-  need_entry = action == GTK_FILE_CHOOSER_ACTION_SAVE ||
-               action == GTK_FILE_CHOOSER_ACTION_CREATE_FOLDER;
-
-  if (need_entry && !dialog->priv->has_entry)
-    {
-      GtkWidget *box;
-      GtkWidget *label;
-      GtkWidget *entry;
-
-      box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
-      label = gtk_label_new_with_mnemonic (_("_Name"));
-      entry = _gtk_file_chooser_entry_new (FALSE, FALSE);
-      g_object_set (label, "margin-start", 6, "margin-end", 6, NULL);
-      g_object_set (entry, "margin-start", 6, "margin-end", 6, NULL);
-      gtk_label_set_mnemonic_widget (GTK_LABEL (label), entry);
-      gtk_container_add (GTK_CONTAINER (box), label);
-      gtk_container_add (GTK_CONTAINER (box), entry);
-      gtk_widget_show_all (box);
-
-      gtk_header_bar_set_custom_title (GTK_HEADER_BAR (header), box);
-      gtk_file_chooser_widget_set_save_entry (GTK_FILE_CHOOSER_WIDGET (dialog->priv->widget), entry);
-    }
-  else if (!need_entry && dialog->priv->has_entry)
-    {
-      gtk_header_bar_set_custom_title (GTK_HEADER_BAR (header), NULL);
-      gtk_file_chooser_widget_set_save_entry (GTK_FILE_CHOOSER_WIDGET (dialog->priv->widget), NULL);
-    }
-
-  dialog->priv->has_entry = need_entry;
-}
-
-static void
 ensure_default_response (GtkFileChooserDialog *dialog)
 {
   GtkWidget *widget;
@@ -577,7 +523,6 @@
   GtkFileChooserDialogPrivate *priv = dialog->priv;
 
   setup_search (dialog);
-  setup_save_entry (dialog);
   ensure_default_response (dialog);
 
   _gtk_file_chooser_embed_initial_focus (GTK_FILE_CHOOSER_EMBED (priv->widget));
